packages/client.api/src/object/ObjectMetadata.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+type ObjectTypeVisibility = "NORMAL" | "PROMINENT" | "HIDDEN";
+
+type BlueprintIcon = {
+  color: string;
+  name: string;
+};
+
+type Icon = BlueprintIcon;
+
+export type ObjectMetadata = {
+  description?: string;
+  displayName: string;
+  icon: Icon;
```

Comment:
Are we guaranteed to always get an icon here? What if its a type we can't support and we want to return undefined in that case?

Reply:
If I understand what you are saying because Icon is a union of just Blueprint right now we are guaranteed to get just a blueprint icon and the types match up. If they dont, ts will throw a flag and we can add the type to our own union

Reply:
Yeh if they add another type this version can keep working because we can just return undefined instead of the wire value until we update the code.

Reply:
Cool yeah I understand, added a check here to conditionally return undefined